### PR TITLE
HSEARCH-2774 The default build should include documentation and javad…

### PR DESCRIPTION
**Pull request by Sanne**
…oc validation

 - https://hibernate.atlassian.net/browse/HSEARCH-2774

### CODE REVIEW

File: documentation/pom.xml [hunk 1]
```diff
@@ -181,6 +181,9 @@
     <profiles>
         <profile>
             <id>documentation-pdf</id>
+            <activation>
+                <activeByDefault>!skipPdfDocumentation</activeByDefault>
```
Comment (yrodiere) [inline documentation/pom.xml]: This is not a good idea. `activeByDefault` means "active if no other profile is enabled" (don't ask me why, just another maven weirdness).
In our case, we have other profiles, so this change didn't make the documentation-pdf profile activate by default.

You may want to use this instead:

```
            <activation>
                <property>
                    <name>!skipPdfDocumentation</name>
                </property>
            </activation>
```

Comment (Sanne) [inline documentation/pom.xml]: oops yes I realized that when testing as I initially used just `activeByDefault`. In fact what I committed later is illegal, activeByDefault doesn't expect a property..

**Comment by Sanne** [comment]
pushed a new version, and changed the scope as commented on JIRA.

**Comment by yrodiere** [comment]
Jenkins, retest this please

**Pull request merged**